### PR TITLE
support admin api extends auth functions

### PR DESCRIPTION
**Pull request by nejisama**
1. Support a new way to registered enhanced admin api functions. keep compatbile for original ways.
2. Export the admin api functions

### CODE REVIEW

File: pkg/admin/server/handler.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+	"mosn.io/mosn/pkg/log"
+)
+
+// APIHandler is a wrapper of http.Handler, which contains auth options
+type APIHandler struct {
+	// the real handler function
+	handler func(http.ResponseWriter, *http.Request)
+	// chains for auth, one of the auth returns false, the request will be failed.
+	auths []func(*http.Request) bool
```
Comment (taoyuanyuan) [inline pkg/admin/server/handler.go:31]: One handler belongs to one user， so I think that no need to use chains.

Comment (taoyuanyuan) [reply pkg/admin/server/handler.go:31]: maybe `auths` and `filead` could combine into one function.
`auths func(w http.ResponseWriter, r *http.Request) bool`?

Comment (nejisama) [reply pkg/admin/server/handler.go:31]: auths functions can be minimized. for example, different auth functions can be combined for different handler

File: pkg/admin/server/handler.go [hunk 2]
```diff
@@ -23,25 +23,49 @@ import (
...
 type APIHandler struct {
 	// the real handler function
 	handler func(http.ResponseWriter, *http.Request)
-	// chains for auth, one of the auth returns false, the request will be failed.
-	auths []func(*http.Request) bool
-	// write repsonse when the request failed by auth function
-	// default is 403 Forbidden without body
-	failed func(http.ResponseWriter)
+	// chains for auth action, different auth can be combined for different handler.
+	// one of the auths check returns false, the
```
Comment (taoyuanyuan) [inline pkg/admin/server/handler.go:58]: Incomplete comment.

Comment (nejisama) [reply pkg/admin/server/handler.go]: fixed

File: pkg/admin/server/handler.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	checkAction  func(*http.Request) bool
+	failedAction func(http.ResponseWriter)
+}
+
+func NewAuth(check func(*http.Request) bool, failed func(http.ResponseWriter)) *Auth {
+	if failed == nil {
+		failed = defaultFailedFunc
+	}
+	return &Auth{
+		checkAction:  check,
```
Comment (taoyuanyuan) [inline pkg/admin/server/handler.go:40]: judge if `check` is nil

Comment (nejisama) [reply pkg/admin/server/handler.go:40]: check is required，i think tif check is nil causes panic is acceptable.

Comment (taoyuanyuan) [reply pkg/admin/server/handler.go:40]: should panic at start up，not at running.

**Review by taoyuanyuan — APPROVED** [review]
LGTM

**Review by wangfakang — APPROVED** [review]

**Pull request merged**